src/resolve-config.js

```diff
@@ -1,31 +1,76 @@
 "use strict";
 
 const cosmiconfig = require("cosmiconfig");
+const editorconfig = require("editorconfig");
+const mem = require("mem");
 const minimatch = require("minimatch");
 
 const asyncWithCache = cosmiconfig("prettier");
 const asyncNoCache = cosmiconfig("prettier", { cache: false });
 const syncWithCache = cosmiconfig("prettier", { sync: true });
 const syncNoCache = cosmiconfig("prettier", { cache: false, sync: true });
 
+const editorconfigAsyncNoCache = filePath =>
+  filePath && editorconfig.parse(filePath).then(editorConfigToPrettier);
+const editorconfigAsyncWithCache = mem(editorconfigAsyncNoCache);
+const editorconfigSyncNoCache = filePath =>
+  filePath && editorConfigToPrettier(editorconfig.parseSync(filePath));
+const editorconfigSyncWithCache = mem(editorconfigSyncNoCache);
+
+function editorConfigToPrettier(editorConfig) {
+  const result = {};
+
+  if (editorConfig.indent_style) {
+    result.useTabs = editorConfig.indent_style === "tab";
+  }
+
+  const tabWidth = editorConfig.indent_size || editorConfig.tab_width;
+  if (tabWidth) {
+    result.tabWidth = tabWidth;
+  }
+
+  if (editorConfig.max_line_length) {
+    result.printWidth = editorConfig.max_line_length;
+  }
+
+  return result;
+}
+
 function resolveConfig(filePath, opts) {
   const useCache = !(opts && opts.useCache === false);
-  return (useCache ? asyncWithCache : asyncNoCache)
-    .load(filePath)
-    .then(result => {
-      return !result ? null : mergeOverrides(result.config, filePath);
-    });
+  return Promise.all([
+    (useCache ? asyncWithCache : asyncNoCache).load(filePath),
+    (useCache ? editorconfigAsyncWithCache : editorconfigAsyncNoCache)(filePath)
+  ]).then((arr /* [result, editorConfigged] */) =>
+    mergeEditorConfig(arr[0], filePath, arr[1])
+  );
 }
 
 resolveConfig.sync = (filePath, opts) => {
   const useCache = !(opts && opts.useCache === false);
   const result = (useCache ? syncWithCache : syncNoCache).load(filePath);
-  return !result ? null : mergeOverrides(result.config, filePath);
+  const editorConfigged = (useCache
+    ? editorconfigSyncWithCache
+    : editorconfigSyncNoCache)(filePath);
+  return mergeEditorConfig(result, filePath, editorConfigged);
 };
 
+function mergeEditorConfig(result, filePath, editorConfigged) {
+  if (!filePath) {
+    return null;
+  }
+
+  const config = result && result.config;
+
+  return Object.assign({}, editorConfigged, mergeOverrides(config, filePath));
+}
+
 function clearCache() {
   syncWithCache.clearCaches();
   asyncWithCache.clearCaches();
+
+  editorconfigSyncWithCache.clear();
+  editorconfigAsyncWithCache.clear();
```

Comment:
mem docs say `mem.clear(fn)`

Reply:
It seems we need the `expect(() => clearCache()).not.toThrowError()` test.

Reply:
Perhaps time to add tests for the clearCache function, then?

Reply:
I tried that when I originally added it but I was having issues with jest module mocks so I gave up. We should look at it again.

Reply:
Oops, I added (and fixed) a test to ensure it doesn't throw. Haven't yet gotten to testing that the cache actually clears.